bftengine/src/preprocessor/messages/PreProcessReplyMsg.cpp

```diff
@@ -66,8 +66,13 @@ void PreProcessReplyMsg::setupMsgBody(const char* buf, uint32_t bufLen) {
 
   // Sign hash
   sigManager_->sign((char*)hash.data(), SHA3_256::SIZE_IN_BYTES, body() + headerSize, sigSize);
-  msgSize_ = headerSize + sigSize;
+  memcpy(body() + headerSize + sigSize, cid.c_str(), cid.size());
+  msgBody()->cidLength = cid.size();
+  msgSize_ = headerSize + sigSize + msgBody()->cidLength;
   msgBody()->replyLength = sigSize;
```

Comment:
Hm. Looks like we don't take into account cid. Let's talk about it.

Reply:
During a discussion, it was clarified that replyLength contains only signed pre-executed hash and does not contain cid length.